common/src/main/java/io/homeassistant/companion/android/common/sensors/LastUpdateManager.kt

```diff
@@ -64,21 +65,49 @@ class LastUpdateManager : SensorManager {
         )
 
         val sensorDao = AppDatabase.getInstance(context).sensorDao()
-        val allSettings = sensorDao.getSettings(lastUpdate.id)
-        val intentSettingName = "lastupdate_intent_var1:${allSettings.size}:"
-        val addNewIntent = allSettings.firstOrNull { it.name == SETTING_ADD_NEW_INTENT }?.value ?: "false"
-        val intentSetting = allSettings.firstOrNull { it.name == intentSettingName }?.value ?: ""
-        if (addNewIntent == "true") {
-            if (intentSetting == "") {
-                sensorDao.add(SensorSetting(lastUpdate.id, SETTING_ADD_NEW_INTENT, "false", SensorSettingType.TOGGLE))
-                sensorDao.add(SensorSetting(lastUpdate.id, intentSettingName, intentAction, SensorSettingType.STRING))
+        val allSettings = sensorDao.getSettings(lastUpdate.id).filter { setting ->
+            val shouldBeRemoved = (setting.value == "")
+            if (shouldBeRemoved) {
+                sensorDao.removeSetting(lastUpdate.id, setting.name)
             }
-        } else {
-            sensorDao.add(SensorSetting(lastUpdate.id, SETTING_ADD_NEW_INTENT, "false", SensorSettingType.TOGGLE))
+            !shouldBeRemoved
         }
-        for (setting in allSettings) {
-            if (setting.value == "")
-                sensorDao.removeSetting(lastUpdate.id, setting.name)
+        var isSequenceContinuous = true
+        var currentHighestIntentSettingOrdinal = 0
+        val intentSettings = allSettings.filter {
+            it.name.startsWith(INTENT_SETTING_PREFIX)
+        }
+        intentSettings.forEachIndexed { index, setting ->
+            val currentOrdinal = setting.name.removePrefix(INTENT_SETTING_PREFIX).removeSuffix(":").toInt()
+            if (currentOrdinal != index + 1) {
+                isSequenceContinuous = false
+            }
+            if (currentOrdinal > currentHighestIntentSettingOrdinal) {
+                currentHighestIntentSettingOrdinal = currentOrdinal
+            }
+        }
+        if (!isSequenceContinuous) {
+            // create new settings with sequential IDs:
+            val newIntentSettings = intentSettings.mapIndexed { index, it ->
+                it.copy(name = "$INTENT_SETTING_PREFIX${index + 1}:")
+            }
+            // delete old settings from DB:
+            intentSettings.forEach {
+                sensorDao.removeSetting(lastUpdate.id, it.name)
+            }
+            // add new settings to DB:
+            newIntentSettings.forEach(sensorDao::add)
+        }
+        val shouldAddNewIntent = allSettings.firstOrNull { it.name == SETTING_ADD_NEW_INTENT }?.value == "true"
+        if (shouldAddNewIntent) {
+            val newIntentSettingOrdinal = currentHighestIntentSettingOrdinal + 1
```

Comment:
`currentHighestIntentSettingOrdinal` isn't updated if the settings are renumbered, which might give a number that is too high instead of the expected value. For example: if there were settings for 1, 3, 4, it will be renumbered to 1, 2, 3, so the expected new setting should be 4 but it will be 5 (`currentHighestIntentSettingOrdinal` = 4).